pyfa_tool/collection.py

```diff
@@ -0,0 +1,201 @@
+            A list of FaDatasets. The default is [].
+
+        Returns
+        -------
+        None.
+
+        """
+        self.ds = None
+        self.FaDatasets = FaDatasets
+
+    # =========================================================================
+    # Specials
+    # =========================================================================
+    def __repr__(self):
+        """String representation."""
+        return str(self.ds)
+
+    def __str__(self):
+        """String typecaste representation."""
+        return str(self.ds)
+
+    # =============================================================================
+    # Getters/setters
+    # =============================================================================
+
+    def set_fadatasets(self, FaDatasets=[]):
+        """
+        Update the FaDatasets of this collection.
+
+        Parameters
+        ----------
+        FaDatasets : list, optional
+            list of (multiple) FaDatasets to add to the collection. The default
+            is [].
+
+        Returns
+        -------
+        None.
+
+        """
+        if len(FaDatasets) == 0:
+            sys.exit('No FaDatasets are provided.')
+        if len(FaDatasets) < 2:
+            sys.exit(f'Only one FaDatasets is provided: {FaDatasets[0]}')
+
+        # type testing
+        for dataset in FaDatasets:
+            if not isinstance(dataset, FaDatasetClass):
+                sys.exit(f'{dataset} is not an instance of FaDataset.')
+
+        # TODO: check for duplicates here
+
+        self.FaDatasets = FaDatasets
+
+    def set_fadatasets_by_file_regex(self, searchdir, filename_regex='*', **kwargs):
+        """
+        Update the FaDatasets of this collection by using regex expression of filenames.
+
+        This methods will select the FA files, according to the regex expression,
+        import the data and update the datasets of this collection.
+
+        Parameters
+        ----------
+        searchdir : str
+            Path to the directory where the regex search query is executed.
+            This is most often the direcotry where the FA files are stored.
+        filename_regex : str, optional
+            Regex expression to match filenames. The default is '*'.
+        **kwargs :
+            kwargs passed to the FaDataset.import_fa() method to specify which
+            fields are imported.
+
+        Returns
+        -------
+        None.
+
+        """
+        # Get paths to the FA files
+        filepaths = IO.get_paths_using_regex(searchdir=searchdir,
+                                             filename_regex=filename_regex)
+        # Read the FaFiles
+        fadatasets = []
+        for file in filepaths:
+            Dataset = FaDatasetClass(fafile=file)
+            Dataset.import_fa(**kwargs)
+            fadatasets.append(Dataset)
+
+        # Add them as attribute
+        self.set_fadatasets(FaDatasets=fadatasets)
+
+    def combine_by_validate(self):
```

Comment:
It feels cumbersome to have to combine a collection manually. I understand that it provides additional functionality if a user would like to combine along a different dimension but maybe a default option in both set_fadatasets (combine=True) which automatically combines on validate would be nice. This can also be used to specify on what to combine if/when combine_by is extended with other coordinates/dimensions. @vergauwenthomas if you agree I can add this.

Reply:
For climate applications i totally agree, but not for nwp. For nwp, `combine_by_validate` will not (or not often) be used. But your suggestion to put `combine` in as argument in the `__init__` is good for me, with a default of False imo.